### PR TITLE
Enhacements

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] Fix tests.
- [x] Switch Psalm.
- [x] Add icon to dropdown menu.

### CODE REVIEW

File: src/Dropdown.php [hunk 1]
```diff
@@ -295,6 +295,19 @@ private function renderItems(array $items, array $itemsOptions = []): string
                 $label = $item['label'];
             }
 
+            $icon = '';
+            $iconOptions = [];
+
+            if (isset($item['icon'])) {
```
Comment (samdark) [inline src/Dropdown.php:301]: You can do `$icon = $item['icon'] ?? ''` instead.

File: src/Dropdown.php [hunk 2]
```diff
@@ -295,6 +295,19 @@ private function renderItems(array $items, array $itemsOptions = []): string
...
+                $icon = $item['icon'];
+            }
+
+            if (isset($item['iconOptions']) && is_array($item['iconOptions'])) {
```
Comment (samdark) [inline src/Dropdown.php:305]:
```suggestion
            if (array_key_exists('iconOptions', $item) && is_array($item['iconOptions'])) {
```

File: src/Widget.php [hunk 3]
```diff
@@ -72,9 +72,9 @@ protected function getId(): ?string
      * @param array $options
      * @param string $valueDefault
      *
-     * @return array
+     * @return array|null
      */
-    protected function addOptions(array $options, string $valueDefault): array
+    protected function addOptions(array $options, string $valueDefault): ?array
```
Comment (samdark) [inline src/Widget.php:77]: Looks wrong. Why can it be null?

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Dropdown.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Dropdown.php]: done.

File: src/Dropdown.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Dropdown.php]: done.

File: src/Widget.php (hunk 3 above)
Comment (terabytesoftw) [reply src/Widget.php]: fixed

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**